src/hooks/useSourceApi.ts

```diff
@@ -260,7 +260,7 @@ const useEditSourceApi = (
   );
 
   const editSources = useCallback(
-    async (payload: SourceType) => {
+    async (payload: SourceRequest & { id: number }) => {
```

Comment:
I've seen this `Request & { id: number }` pattern in a few other places in this PR. What exactly does that do? I see that the definition of `type SourceRequest` has `id?: number;` so I probably just don't understand why it's also being set here.